File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(num):
	# https://www.google.de/search?q=sum+of+sequence
    seq_sum = num * (1 + num) / 2

    return seq_sum * seq_sum


def sum_of_squares(num):
	# http://www.trans4mind.com/personal_development/mathematics/series/sumNaturalSquares.htm
    return num * (num + 1) * (2 * num + 1) / 6

def difference(num):
    return square_of_sum(num) - sum_of_squares(num)

File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum (number: int): int =
    List.sum [1 .. number] |> (fun x -> x * x)

let sumOfSquares (number: int): int =
    List.sumBy (fun x -> x * x) [1 .. number]

let differenceOfSquares (number: int): int =
    squareOfSum number - sumOfSquares number
